google_guest_agent/main.go

```diff
@@ -283,9 +294,13 @@ func main() {
 		action = os.Args[1]
 	}
 
-	if action == "noservice" {
+	switch action {
+	case "noservice":
 		runAgent(ctx)
 		os.Exit(0)
+	case "get_config_option":
```

Comment:
is this used only to query the configuration?

Reply:
Yes

Reply:
That will work and be simpler, but if we expand the socket-based communication to cover more than just the network as we discussed this would no longer work.